### DISCUSSION TITLE
rotationPercent and rotationAbsolute resets after page change or screen lock

### DISCUSSION THREAD

**Discussion by artistro08** [cgiesche/streamdeck-homeassistant#275] [Help and Support] [answered]
So I have a volume knob for one of my media devices

rotating them to change the volume works amazing.

But keeping that volume is an issue. Since both `rotationPercent` and `rotationAbsolute` reset their values (i'm presuming) they set the volume to 0. 

If we could get the volume variable in the json area, that would be a temporary fix, as I can use the ticks to adjust. 

Here's my code for reference:

```json
{ "volume_level": {{ rotationAbsolute / 255 }} }
```

**Comment by cgiesche** [comment, marked as answer]
Hi,

i understand your wish to have access to state variables in the service call JSON. Long story short: It would be very complicated to implement and make configuration much more complex than it already is. 

But there is hope:

```json
{
  "color_temp": "{% raw %}{{ state_attr('light.living_room_ikea','color_temp') + {% endraw %}{{ ticks }}{% raw %} }}{% endraw %}"
}
```

Not only my plugin supports the nunjucks syntax, the home-assistant side does as well. To send (parts) of the template to homeassistant, you can use the `raw` option.

The above example sets the `color_temp` to the current (home-assistant side) value + the plugin's side ticks value. It looks confusing, but you should be able to use this to do exactly what you need :)

Entered in plugin:
```json
{
  "color_temp": "{% raw %}{{ state_attr('light.living_room_ikea','color_temp') + {% endraw %}{{ ticks }}{% raw %} }}{% endraw %}"
}
```

Sent to home-assistant after non-raw values were rendered on plugin side:
```json
{
  "color_temp": "{{ state_attr('light.living_room_ikea','color_temp') + 5 }}"
}
```

This is rendered again on home-assistant server side (where you can do almost everything).

**Comment by artistro08** [comment]
That's awesome! 

I feel like I'm closer, but not close enough. 

I tried this: 

```json

{ "volume_level": "{% raw %}{{ state_attr('media_player.chromecastaudio6417','volume_level') + {% endraw %}{{ ticks / 100 }}{% raw %} }}{% endraw %}"}

```


And it's not quite working.

**Reply by artistro08** [reply]
Nevermind, Got it working! 

Code for reference:

```json
{ "volume_level": "{% raw %}{{ state_attr('media_player.office_speakers','volume_level') + {% endraw %}{{ ticks / 100 }}{% raw %} }}{% endraw %}"}
```

**Comment by artistro08** [comment]
Sorry to awaken an old thread but how do I make it so the light doesn't go off when it hits zero? My math escapes me

**Reply by artistro08** [reply]
Nevermind, figured this out too! 

Using `brightness_step` and some `if else` statements made it easy. 

```json
{ "brightness_step": "{% raw %}{% if (state_attr('light.desk_key_light','brightness') + {% endraw %}{{ ticks }}{% raw %}) <= 10 %} 0{% else %} {{ {% endraw %}{{ ticks }}{% raw %} }}{% endif %}{% endraw %}"}
```

**Comment by artistro08** [comment]
Thanks again for your help!
